src/python/pants/jvm/bsp/spec.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from pants.bsp.spec import Uri
+
+
+@dataclass(frozen=True)
+class JvmBuildTarget:
+    # Uri representing absolute path to jdk
+    # For example: file:///usr/lib/jvm/java-8-openjdk-amd64
+    java_home: Uri | None = None
+
+    # The java version this target is supposed to use.
+    # For example: 1.8
+    java_version: str | None = None
```

Comment:
Why are these optional?

Reply:
Because they are optional in the BSP spec. This is a direct binding of the BSP specification.

Reply:
See https://build-server-protocol.github.io/docs/extensions/jvm.html#jvm-build-target